Introduce the terms 'Proposal' and 'Commit' (#200)
mlswg/mls-architecture: draft-ietf-mls-architecture.md

then it is the responsibility of the application to inform members
of this policy and who the administrator is.

## Proposals and Commits

The general pattern for any change in the group state (e.g., to add or remove
a user) is that it consists of two messages:

Proposal
: This message describes the change to be made (e.g., add Bob to the group)
but does not effect a change.

Commit
: This message changes the group state to include the changes described in
a set of proposals.

> The simplest pattern is for a client to just send a Commit which contains one or more Proposals, for instance Alice could send the a Commit with the Proposal Add(Bob) embedded to add

Reviewer: ```suggestion
The simplest pattern is for a client to just send a Commit which contains one or more Proposals,
for instance Alice could send a Commit with the Proposal Add(Bob) embedded to add
```

s/the a Commit/a Commit/

Bob to the group. However, there are situations in which one client
might send a proposal and another might send the commit. For instance,

> Bob might wish to remove himself the group and send a Remove Proposal to do so (see

Reviewer: ```suggestion
Bob might wish to remove himself from the group and send a Remove Proposal to do so (see
```

then it is the responsibility of the application to inform members
of this policy and who the administrator is.

## Proposals and Commits

The general pattern for any change in the group state (e.g., to add or remove
a user) is that it consists of two messages:

Proposal
: This message describes the change to be made (e.g., add Bob to the group)
but does not effect a change.

Commit
: This message changes the group state to include the changes described in
a set of proposals.

> The simplest pattern is for a client to just send a single Proposal/Commit
> pair, for instance Alice could send the pair Add(Bob), Commit to add

Reviewer: ```suggestion
The simplest pattern is for a client to just send a Commit which contains one or more Proposals, for instance Alice could send the a Commit with the Proposal Add(Bob) embedded to add
```

Bob to the group. However, there are situations in which one client
might send a proposal and another might send the commit. For instance,

> Bob might wish to join the group and send a Proposal to do so (see
> {{Section 12.1.6 of ?RFC9420}}). Because Bob is not already a group member
> he cannot send the Commit and an existing member must do so.

Reviewer: "[Bob] cannot send the Commit" is confusing because External Commits do exactly that. Perhaps better to use the case of leaving a group?

```suggestion
Bob might wish to remove himself the group and send a Remove Proposal to do so (see
{{Section 12.1.3 of ?RFC9420}}). Because Bob
cannot send the Commit, an existing member must do so.
```

> Commits can apply to multiple Proposals, in which case all the listed

Reviewer: ```suggestion
Commits can apply to multiple valid Proposals, in which case all the listed
```

changes are applied.

It is also possible for a Commit to apply to an empty set of Proposals
in which case it just updates the cryptographic state of the group

> without changing it's membership.

Reviewer: ```suggestion
without changing its membership.
```
